deepset_cloud_sdk/_s3/upload.py

```diff
@@ -283,13 +283,13 @@ async def upload_texts(
             for file in files:
                 # raw data
                 file_name = file.name
-                tasks.append(self.upload_from_string(file_name, upload_session, file.text, client_session))
+                tasks.append(self.upload_from_memory(file_name, upload_session, file.content(), client_session))
 
                 # meta
                 if file.meta is not None:
                     meta_name = f"{file_name}.meta.json"
-                    metadata = json.dumps(file.meta)
-                    tasks.append(self.upload_from_string(meta_name, upload_session, metadata, client_session))
+                    metadata = bytes(json.dumps(file.meta), encoding="utf-8")
```

Comment:
why does the metadata need to be bytes too ? can't we just keep the dict here and convert it at the last step ?

Reply:
even previously we were converting to string, but the upload from memory step only takes bytes. Will update for it to accept string too.